store/sqlstore/post_store.go

```diff
@@ -2002,7 +2002,12 @@ func (s *SqlPostStore) search(teamId string, userId string, params *model.Search
 			termsClause = "(" + strings.Join(strings.Fields(terms), " & ") + ")" + excludeClause
 		}
 
-		searchClause := fmt.Sprintf("to_tsvector('%[1]s', %[2]s) @@  to_tsquery('%[1]s', ?)", s.pgDefaultTextSearchConfig, searchType)
+		// Similar to the "terms", replace special chars with space in the source string
+		specialChars := strings.Join(specialSearchChar, "")
+		replaceWith := strings.Repeat(" ", len(specialSearchChar))
+		translatedSrc := fmt.Sprintf("translate(%s, '%s', '%s')", searchType, specialChars, replaceWith)
+
+		searchClause := fmt.Sprintf("to_tsvector('%[1]s', %[2]s) @@  to_tsquery('%[1]s', ?)", s.pgDefaultTextSearchConfig, translatedSrc)
```

Comment:
This most probably interferes with the GIN indexes we have in place for the `hashtags` and `message` columns:
```
"idx_posts_hashtags_txt" gin (to_tsvector('english'::regconfig, hashtags::text))
"idx_posts_message_txt" gin (to_tsvector('english'::regconfig, message::text))
```
For a full text-search query like this to use these GIN indexes, the `tsvector` crafted here must match the structure of the GIN index itself, `translate` calls included.

Could you provide an example of a resulting query from this code so we can check whether it would use the index or not?

Reply:
@agarciamontoro You're right, `translate` doesn't use the index. 
Now, I've added the "@" exception just for Postgres, as it can handle it well.

Reply:
Here are the 2 queries I tried with
```sql
select id, message  from posts p 
where to_tsvector('english', translate(message, '@', ' ')) @@ to_tsquery('(\"test <-> test.com\")');

select id, message  from posts p 
where to_tsvector('english', message) @@ to_tsquery('(\"test <-> test.com\")');
```

Reply:
GIN indexes are a bit of a nitpickers :unamused: